src/utils/source-mapper.ts

```diff
@@ -0,0 +1,103 @@
+import type { elementNode, serializedNodeWithId } from '@chromaui/rrweb-snapshot';
+import { NodeType } from '@chromaui/rrweb-snapshot';
+import path from 'path';
+import { logger } from './logger';
+
+/**
+ * A builder class that allows us to transform "src" entries within dom snapshots output
+ * by `rrweb-snapshot`. This is done for post-processing of the data to avoid issues
+ * where the "src" entry would not work well with the underlying file system.
+ */
+export class SourceMapper {
+  private domSnapshot: elementNode;
+
+  private shortenedFileNameLength: number;
+
+  /**
+   * Create a new {@link SourceMapper} object at a given root node.
+   *
+   * @param domSnapshot The {@link elementNode} where processing should start. This can be the dom
+   * snapshot that's retrieved when `rrweb-snapshot` is first invoked.
+   */
+  constructor(domSnapshot: elementNode) {
+    this.domSnapshot = domSnapshot;
+    this.shortenedFileNameLength = -1;
+  }
+
+  /**
+   * Specify a maximum length, in bytes, that any individual source path component can be.
+   *
+   * Note: Since "src" entries are assumed to be URIs (either in absolute or relative form),
+   * the separator for individual path components will always be '/'.
+   *
+   * @param maxLength A {@link number} that specifies the maximum number of bytes that can
+   * be used for each individual path component within an initial "src" entry.
+   *
+   * @returns `this`, the {@link SourceMapper} object upon which the method was called, for chaining.
+   */
+  shortenFileNamesLongerThan(maxLength: number): SourceMapper {
+    this.shortenedFileNameLength = maxLength;
+
+    return this;
+  }
```

Comment:
How about passing in the `maxLength` into the constructor, and thus saving having to call this method?

Reply:
I thought about that. The reason I didn't do that is because I want to emphasize the builder pattern for future additions. Right now, this is the only real method for this, but as time goes on, we'll likely have more of these methods and I wanted to have an example of how to add to it. Further, it's entirely possible that we might have another method, say something like, `replaceOneWithAnother(x: string, y: string)`, and call it in a manner like the following:
```
let sourceMapper = new SourceMapper(domSnapshot);
let sourceMap = sourceMapper.replaceOneWithAnother(x, y).build();
```

In that example, the method `shortenFileNamesLongerThan()` isn't called at all, but if we require it in the constructor, then we'd have to do that part of the process.